Assets/MixedRealityToolkit-SDK/Features/Diagnostics/MixedRealityDiagnosticsManager.cs

```diff
@@ -0,0 +1,243 @@
+        private static readonly ExecuteEvents.EventFunction<IMixedRealityDiagnosticsHandler> OnDiagnosticsChanged =
+            delegate (IMixedRealityDiagnosticsHandler handler, BaseEventData eventData)
+            {
+                DiagnosticsEventData diagnosticsEventsData = ExecuteEvents.ValidateEventData<DiagnosticsEventData>(eventData);
+                handler.OnDiagnosticSettingsChanged(diagnosticsEventsData);
+            };
+
+        #endregion
+
+        #region IMixedRealityDiagnosticsManager
+        private bool visible;
+
+        /// <inheritdoc />
+        public bool Visible
+        {
+            get
+            {
+                return visible;
+            }
+
+            set
+            {
+                if (value != visible)
+                {
+                    visible = value;
+                    GetDiagnosticVisualization()?.SetActive(value);
+
+                    RaiseDiagnosticsChanged();
+                }
+            }
+        }
+
+        private bool showCpu;
+
+        /// <inheritdoc />
+        public bool ShowCpu
+        {
+            get
+            {
+                return showCpu;
+            }
+
+            set
+            {
+                if (value != showCpu)
+                {
+                    showCpu = value;
+                    var component = GetDiagnosticVisualization()?.GetComponent<DiagnosticBehavior>();
+                    if (component != null)
+                    {
+                        component.ShowCpu = value;
+                    }
+
+                    RaiseDiagnosticsChanged();
+                }
+            }
+        }
+
+        private bool showFps;
+
+        /// <inheritdoc />
+        public bool ShowFps
+        {
+            get
+            {
+                return showFps;
+            }
+            set
+            {
+                if (value != showFps)
+                {
+                    showFps = value;
+                    var component = GetDiagnosticVisualization()?.GetComponent<DiagnosticBehavior>();
+                    if (component != null)
+                    {
+                        component.ShowFps = value;
+                    }
+
+                    RaiseDiagnosticsChanged();
+                }
+            }
+        }
+
+        private bool showMemory;
+
+        /// <inheritdoc />
+        public bool ShowMemory
+        {
+            get
+            {
+                return showMemory;
+            }
+            set
+            {
+                if (value != showMemory)
+                {
+                    showMemory = value;
+                    var component = GetDiagnosticVisualization()?.GetComponent<DiagnosticBehavior>();
+                    if (component != null)
+                    {
+                        component.ShowMemory = value;
+                    }
+
+                    RaiseDiagnosticsChanged();
+                }
+            }
+        }
+        #endregion
```

Comment:
let's add the name of the region to the end as well please

`#endregion IMixedRealityDiagnosticsManager`